mysql/datadog_checks/mysql/activity.py

```diff
@@ -0,0 +1,292 @@
+import datetime
+import decimal
+import time
+from contextlib import closing
+from enum import Enum
+from typing import Dict, List
+
+import pymysql
+
+from datadog_checks.base import is_affirmative, to_native_string
+from datadog_checks.base.utils.db.sql import compute_sql_signature
+from datadog_checks.base.utils.db.utils import DBMAsyncJob, obfuscate_sql_with_metadata
+from datadog_checks.base.utils.serialization import json
+from datadog_checks.base.utils.tracking import tracked_method
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+CONNECTIONS_QUERY = """\
+SELECT
+    processlist_user,
+    processlist_host,
+    processlist_db,
+    processlist_state,
+    COUNT(processlist_user) AS connections
+FROM
+    performance_schema.threads
+WHERE
+    processlist_user IS NOT NULL AND
+    processlist_state IS NOT NULL
+    GROUP BY processlist_user, processlist_host, processlist_db, processlist_state
+"""
+
+ACTIVITY_QUERY = """\
+SELECT
+    thread_a.thread_id,
+    thread_a.processlist_id,
+    thread_a.processlist_user,
+    thread_a.processlist_host,
+    thread_a.processlist_db,
+    thread_a.processlist_command,
+    thread_a.processlist_state,
+    thread_a.processlist_info AS sql_text,
+    statement.timer_start AS event_timer_start,
+    statement.timer_end AS event_timer_end,
+    statement.lock_time,
+    statement.current_schema,
+    COALESCE(
+        IF(thread_a.processlist_state = 'User sleep', 'User sleep',
+        IF(waits_a.event_id = waits_a.end_event_id, 'CPU', waits_a.event_name)), 'CPU') AS wait_event,
+    waits_a.event_id,
+    waits_a.end_event_id,
+    waits_a.event_name,
+    waits_a.timer_start AS wait_timer_start,
+    waits_a.timer_end AS wait_timer_end,
+    waits_a.object_schema,
+    waits_a.object_name,
+    waits_a.index_name,
+    waits_a.object_type,
+    waits_a.source,
+    socket.ip,
+    socket.port
+FROM
+    performance_schema.threads AS thread_a
+    LEFT JOIN performance_schema.events_waits_current AS waits_a ON waits_a.thread_id = thread_a.thread_id AND
+    waits_a.event_id IN(
+        SELECT
+            MAX(waits_b.EVENT_ID)
+        FROM performance_schema.threads AS thread_b
+            LEFT JOIN performance_schema.events_waits_current AS waits_b ON waits_b.thread_id = thread_b.thread_id
+        WHERE
+            thread_b.processlist_state IS NOT NULL AND
+            thread_b.processlist_command != 'Sleep' AND
+            thread_b.processlist_command != 'Daemon' AND
+            thread_b.processlist_id != connection_id()
+        GROUP BY thread_b.thread_id)
+    LEFT JOIN performance_schema.events_statements_current AS statement ON statement.thread_id = thread_a.thread_id
+    LEFT JOIN performance_schema.socket_instances AS socket ON socket.thread_id = thread_a.thread_id
+WHERE
+    thread_a.processlist_state IS NOT NULL AND
+    thread_a.processlist_command != 'Sleep' AND
+    thread_a.processlist_command != 'Daemon' AND
```

Comment:
What is this excluding?

Reply:
filtering out idle threads. If the thread is sleeping or state is NULL then it is idle.
The filter on Daemon is questionable.  We want to show background, i.e. non-user / non-application waits and activity, but in my testing Daemon entries looked idle. Would be worth investigating more.

Reply:
here is output with "daemon" 

    +-----------+--------+------------------------+----------+
    | thread_id | cmd    | PROCESSLIST_STATE      | EVENT_ID | 
    +-----------+--------+------------------------+----------+
    |        42 | Daemon | Waiting on empty queue |     NULL |
    |        46 | Daemon | Suspending             |     NULL |

Guess we could try filtering on COMMMAND and PROCESSLIST_STATE but I don't have a list of PROCESSLIST_STATE values go filter. We could collect them and fitler on the backend.

Reply:
> We could collect them and fitler on the backend.
+1, we should submit everything we think we might need because we can always adjust things later in our back-end.

Reply:
I would bias towards collecting fewer rows until we need them. But in this case it doesn't seem that we're confident daemon threads are truly idle. 

```
Daemon

This thread is internal to the server, not a thread that services a client connection.
```

https://dev.mysql.com/doc/refman/8.0/en/thread-commands.html

```
COMMAND

The type of command the thread is executing on behalf of the client, or Sleep if the session is idle.
```
https://dev.mysql.com/doc/refman/8.0/en/performance-schema-processlist-table.html


It's hard to say but the language seems to suggest Daemon is non-idle.